### PR TITLE
Cog 357 purge user log

### PR DESCRIPTION
**Pull request by dmytro-palczewski-osi**
### JIRA Issue Link
- [COG-357: Purge User update/create log table](https://osi-cwds.atlassian.net/browse/COG-357)

### Technical Description
Purge old data from user failed operations log table is implemented

### Tests
- [ ] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/idm/service/UserLogService.java [hunk 1]
```diff
@@ -52,6 +52,10 @@
         .collect(Collectors.toList());
   }
 
+  public int deleteAlreadyProcessedLogs(Date lastJobTime) {
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/idm/service/UserLogService.java]: I would suggest renaming the method and parameter to deleteProcessedLogs(Date beforeDate)

File: src/main/java/gov/ca/cwds/idm/persistence/UserLogRepository.java [hunk 2]
```diff
@@ -22,4 +23,9 @@
           + "group by u.username, u.operationType")
   @Transactional(value = "tokenTransactionManager", readOnly = true)
   List<Object[]> getUserIdAndOperationTypes(@Param("lastJobTime") Date lastJobTime);
+
+  @Query("delete from UserLog u where u.operationTime <= :lastDate")
+  @Modifying
+  @Transactional(value = "tokenTransactionManager")
+  int deleteLogsBeforeLastDate(@Param("lastDate") Date lastDate);
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/idm/persistence/UserLogRepository.java]: Consider renaming to deleteLogsBeforeDate(Date date)

File: src/test/java/gov/ca/cwds/idm/IdmResourceTest.java [hunk 3]
```diff
@@ -868,6 +879,19 @@ private UserLog userLog(String userName, OperationType operation,  long date) {
...
   }
 
+  private void assertUserLog(
+      UserLog userLog, String username, OperationType operationType, long time) {
+    assertThat(userLog.getUsername(), is(username));
+    assertThat(userLog.getOperationType(), is(operationType));
+    assertThat(userLog.getOperationTime(), is(new Date(time)));
+  }
+
+  private void assertUserLog(
```
[the remark this answers is not in the document]
Comment (dmytro-palczewski-osi) [reply src/test/java/gov/ca/cwds/idm/IdmResourceTest.java]: Yes, it's used by another method

File: src/main/java/gov/ca/cwds/idm/persistence/UserLogRepository.java (hunk 2 above)
Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/idm/persistence/UserLogRepository.java]: renamed

File: src/main/java/gov/ca/cwds/idm/service/UserLogService.java (hunk 1 above)
Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/idm/service/UserLogService.java]: renamed

**Pull request merged**